lib/index.js

```diff
@@ -234,73 +234,165 @@ module.exports = function (options, handler) {
 };
 
 function mapResponseHeaders(event, context, result) {
-    const headers = result.headers || {}; // NOTE: Mutating event.headers; prefer deep clone of event.headers
-    return headers;
+  const headers = Object.assign({}, result.headers || {}); // NOTE: Mutating event.headers; prefer deep clone of event.headers
+  return headers;
 }
 
-function logEvent(event, context, err, result, options, moesifController) {
-  if (!event.httpMethod || !event.headers) {
-    logMessage(
-      options.debug,
-      'logEvent',
-      'AWS Lambda trigger must be a Load Balancer or API Gateway. ' +
-        'See https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-set-up-simple-proxy.html#api-gateway-set-up-lambda-proxy-integration-on-proxy-resource'
-    );
-    return Promise.resolve();
+// in V2
+// https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+// if there is no statusCode but valid JSON, API gateway will interpret
+// the response.
+function translateLambdaResultIfNeeded(result, isV1) {
+  if (isV1) {
+    return result || {};
   }
 
-  if (options.skip(event, context)) {
-    // exit early
-    return Promise.resolve();
+  if (result && result.statusCode) {
+    return result;
   }
 
+  if (!result.statusCode && (typeof result === 'object' || typeof result === 'string')) {
```

Comment:
Is it possible to return a null/undefined result in Lambda?

Reply:
I assume so.